x-pack/plugins/apm/public/components/shared/kuery_bar/get_bool_filter.ts

```diff
@@ -35,6 +37,17 @@ export function getBoolFilter({
     });
   }
 
+  const environmentFilter = environmentQuery(urlParams.environment);
+  if (!isEmpty(environmentFilter)) {
```

Comment:
I don't think you need this check, you'll just be pushing an empty array, which is essentially a noop (I assume).